### PR TITLE
BE |  Refactor dynamics service class

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- Created `Dynamics::Service`
- Created `Dynamics::ErrorHandler` (only for the `service`)
- Created `DynamicsMockService`
  - class to handle the mock data before the `API` is created
- Use `DatadogLogger` to log `responses` 

## Related issue(s)
- department-of-veterans-affairs/ask-va#283

## Testing done

- `vets-website` `ask-va` FE should function as is
- need to check `Datadog` in staging for logging
- need to check `Sentry` in staging for logging

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by RachalCassity** [comment]
@Khoa-V-Nguyen Please add the new directories to the .github/CODEOWNERS file.

File: modules/ask_va_api/app/services/dynamics/service.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      raise ErrorHandler::ServiceError, "Error parsing response from #{endpoint}: #{e.message}"
+    end
+
+    def invoke_request(endpoint, method, payload, params)
+      logger.call("api_call.#{method}", tags: build_tags(endpoint)) do
+        conn.public_send(method, endpoint, prepare_payload(method, payload, params))
+      end
+    end
+
+    def prepare_payload(method, payload, params)
```
Comment (jakehova) [inline modules/ask_va_api/app/services/dynamics/service.rb:99]: Should have a default here that raises an error if the method passed in doesn't match one of the ones specified

File: modules/ask_va_api/spec/services/dynamics/service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Dynamics::Service do
```
Comment (jakehova) [inline modules/ask_va_api/spec/services/dynamics/service_spec.rb:5]: add a test for the prepare_payload method that verifies an error gets raised if the method passed in is not get, post, patch, or put

**Review by jakehova — CHANGES_REQUESTED** [review]

File: modules/ask_va_api/app/services/dynamics/service.rb (hunk 1 above)
Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/app/services/dynamics/service.rb:99]: the method validation is being handled in the call

File: modules/ask_va_api/spec/services/dynamics/service_spec.rb (hunk 2 above)
Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/spec/services/dynamics/service_spec.rb:5]: same as above

**Review by joehall-tw — APPROVED** [review]
Read comments and Khoa's replies. Quick scan of the code looks good to me. Approving.

**Review by jakehova — APPROVED** [review]

**Review by jakehova — DISMISSED** [review]

**Pull request merged**